### PR TITLE
feat: Making Pip installer from Pypi

### PR DESCRIPTION
**Pull request by dciborow**
This packages the MMLSpark Python package with the Jar, and makes Python imports possible prior to starting SparkContext.

### CODE REVIEW

File: src/main/python/MANIFEST.in [hunk 1]
```diff
@@ -3,3 +3,4 @@ include LICENSE.txt
 
 # documentation, if any - there may be a minimal set of docs:
 # recursive-include docs *.html *.txt *.js
+recursive-include deps/jars *.jar
```
Comment (akshaya-a) [inline src/main/python/MANIFEST.in]: when does the jar get built in your workflow above?

Comment (dciborow) [reply src/main/python/MANIFEST.in]: I had cheated and uploaded it into a commit. Updated  script to the pipeline so it will pull the jar from there.

**Comment by mhamilton723** [comment]
@dciborow did this ever get resolved on ADB?

File: project/Secrets.scala [hunk 2]
```diff
@@ -45,5 +45,6 @@ object Secrets {
     sys.env.getOrElse("PGP-PRIVATE", getSecret("pgp-private")).getBytes("UTF-8")))
   lazy val pgpPassword: String = sys.env.getOrElse("PGP-PW", getSecret("pgp-pw"))
   lazy val storageKey: String = sys.env.getOrElse("STORAGE_KEY", getSecret("storage-key"))
+  lazy val PyPiPw: String = sys.env.getOrElse("TWINE_PASSWORD", getSecret("pypi-pw"))
```
Comment (mhamilton723) [inline project/Secrets.scala]: nit: lowercase p
```suggestion
  lazy val pyPiPw: String = sys.env.getOrElse("TWINE_PASSWORD", getSecret("pypi-pw"))
```

File: build.sbt [hunk 3]
```diff
@@ -169,6 +169,25 @@ def pythonizeVersion(v: String): String = {
...
+  packagePythonTask.value
+
+  Process(
+    activateCondaEnv ++
+      Seq(s"python", "-m", "pip", "install", "twine"),
+    pythonPackageDir) ! s.log
+
+  Process(
+    activateCondaEnv ++
+      Seq(s"python", "-m", "twine", "upload", "*", "-u", "__token__", "-p", s"${Secrets.PyPiPw}"),
```
Comment (mhamilton723) [inline build.sbt]:
```suggestion
      Seq(s"python", "-m", "twine", "upload", "*", "-u", "__token__", "-p", s"${Secrets.pyPiPw}"),
```

File: build.sbt [hunk 4]
```diff
@@ -169,6 +169,25 @@ def pythonizeVersion(v: String): String = {
   }
 }
 
+val uploadPyPiTask = TaskKey[Unit]("uploadPyPiPackage", "upload python sdk to Pypi")
```
Comment (dciborow) [inline build.sbt]:
```suggestion
val publishPyPi = TaskKey[Unit]("publishPyPi", "upload python sdk to Pypi")
```

File: build.sbt [hunk 5]
```diff
@@ -169,6 +169,25 @@ def pythonizeVersion(v: String): String = {
...
+
+uploadPyPiTask := {
```
Comment (dciborow) [inline build.sbt]:
```suggestion
publishPyPi := {
```

File: pipeline.yaml [hunk 6]
```diff
@@ -59,7 +59,7 @@ jobs:
     - bash: |
         source activate mmlspark
         sbt packagePython
-        sbt publishBlob publishSigned publishDocs publishR
+        sbt publishBlob publishSigned publishDocs publishR uploadPyPiTask
```
Comment (dciborow) [inline pipeline.yaml]:
```suggestion
        sbt publishBlob publishSigned publishDocs publishR publishPyPi
```

File: src/main/python/setup.py [hunk 7]
```diff
@@ -6,12 +6,12 @@
 
 setup(
     name="mmlspark",
-    version=os.environ["MML_PY_VERSION"],
+    version="0.0.11111", #os.environ["MML_PY_VERSION"],
```
Comment (dciborow) [inline src/main/python/setup.py]:
```suggestion
    version=os.environ["MML_PY_VERSION"],
```

**Comment by ekerazha** [comment]
@mhamilton723 Is it possible to merge this before the next release?

**Pull request closed without merging**